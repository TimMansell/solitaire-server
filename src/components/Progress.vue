<template>
  <span class="progress">
    <span class="progress__bar" />
  </span>
</template>

<script>
export default {
  name: 'Progress',
};
</script>

<style scoped lang="scss">
.progress {
  display: flex;

  &__bar {
    flex: 1;
    height: var(--vr);
    border-radius: var(--bdr-radius-lg);

    /* stylelint-disable */

    background-image: linear-gradient(
      90deg,
      var(--col-secondary-alt) 0,
      var(--col-secondary-alt-2) 40px,
      var(--col-secondary-alt) 100px
    );

    /* stylelint-enable */

    background-size: 200%;

    @media screen and (prefers-reduced-motion: no-preference) {
      animation: animate-bar 1.5s infinite linear;
    }
  }
}

@keyframes animate-bar {
  0% {
    background-position: 100%;
  }

  40%,
  100% {
    background-position: -100%;
  }
}
</style>
