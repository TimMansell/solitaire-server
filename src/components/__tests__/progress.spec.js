import { shallowMount } from '@vue/test-utils';
import Progress from '@/components/Progress.vue';

describe('Progress.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(Progress);

    expect(wrapper).toMatchSnapshot();
  });
});
