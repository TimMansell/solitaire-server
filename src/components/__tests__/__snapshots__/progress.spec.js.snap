// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress.vue matches snapshot 1`] = `<span class="progress"><span class="progress__bar"></span></span>`;
